ldpd/ldp_vty_cmds.c

```diff
@@ -0,0 +1,1134 @@
+/*
+ * Copyright (C) 2016 by Open Source Routing.
+ *
+ * This file is part of GNU Zebra.
+ *
+ * GNU Zebra is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2, or (at your option) any
+ * later version.
+ *
+ * GNU Zebra is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; see the file COPYING; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+#include <zebra.h>
+
+#include "command.h"
+#include "vty.h"
+#include "ldpd/ldp_vty.h"
+
+DEFUN_NOSH(ldp_mpls_ldp,
+	ldp_mpls_ldp_cmd,
+	"[no] mpls ldp",
+	"Negate a command or set its defaults\n"
+	"Global MPLS configuration subcommands\n"
+	"Label Distribution Protocol\n")
+{
+	int		 idx = 0;
+	int		 negate = 0;
+
+	if (argv_find(argv, argc, "no", &idx))
+		negate = 1;
+
+	return (ldp_vty_mpls_ldp(vty, negate));
+}
+
+DEFUN_NOSH(ldp_l2vpn,
+	ldp_l2vpn_cmd,
+	"[no] l2vpn WORD type vpls",
+	"Negate a command or set its defaults\n"
+	"Configure l2vpn commands\n"
+	"L2VPN name\n"
+	"L2VPN type\n"
+	"Virtual Private LAN Service\n")
+{
+	int		 idx = 0;
+	int		 negate = 0;
+	const char	*name = NULL;
+
+	if (argv_find(argv, argc, "no", &idx))
+		negate = 1;
+	if (argv_find(argv, argc, "WORD", &idx))
```

Comment:
yep, that's what figured with the `argv_find`. More readable that way and it's O(1) in any case so fine by me :)